code/modules/paperwork/fax_manager.dm

```diff
@@ -0,0 +1,144 @@
+					return TRUE
+		if("flw_fax")
+			for(var/obj/machinery/fax/FAX in GLOB.machines)
+				if (FAX.fax_id == params["fax_id"])
+					usr.client.admin_follow(FAX)
+					return TRUE
+		if("read_message")
+			var/list/REQUEST = get_request(params["id_message"])
+			var/obj/item/paper/request/paper = REQUEST["paper"]
+			paper.ui_interact(usr)
+			return TRUE
+		if("flw")
+			var/list/REQUEST = get_request(params["id_message"])
+			usr.client.admin_follow(REQUEST["sender"])
+			return TRUE
+		if("pp")
+			var/list/REQUEST = get_request(params["id_message"])
+			usr.client.holder.show_player_panel(REQUEST["sender"])
+			return TRUE
+		if("vv")
+			var/list/REQUEST = get_request(params["id_message"])
+			usr.client.debug_variables(REQUEST["sender"])
+			return TRUE
+		if("sm")
+			var/list/REQUEST = get_request(params["id_message"])
+			usr.client.cmd_admin_subtle_message(REQUEST["sender"])
+			return TRUE
+		if("logs")
+			var/list/REQUEST = get_request(params["id_message"])
+			if(!ismob(REQUEST["sender"]))
+				to_chat(usr, "This can only be used on instances of type /mob.", confidential = TRUE)
+				return TRUE
+			show_individual_logging_panel(REQUEST["sender"], null, null)
+			return TRUE
+		if("smite")
+			var/list/REQUEST = get_request(params["id_message"])
+			if(!check_rights(R_FUN))
+				to_chat(usr, "Insufficient permissions to smite, you require +FUN", confidential = TRUE)
+				return TRUE
+			var/mob/living/carbon/human/H = REQUEST["sender"]
+			if (!H || !istype(H))
+				to_chat(usr, "This can only be used on instances of type /mob/living/carbon/human", confidential = TRUE)
+				return TRUE
+			usr.client.smite(H)
+			return TRUE
+
+/datum/fax_manager/proc/get_request(id_message)
+	for(var/list/REQUEST in requests)
+		if(REQUEST["id_message"] == id_message)
+			return REQUEST
+
+/datum/fax_manager/proc/receive_request(mob/sender, obj/machinery/fax/sender_fax, receiver_fax_name, obj/item/paper/paper, receiver_color)
+	var/list/request = list()
+	var/obj/item/paper/request/message = new()
+	request["id_message"] = requests.len
+	request["time"] = gameTimestamp()
+	request["sender"] = sender
+	request["sender_fax_id"] = sender_fax.fax_id
+	request["sender_fax_name"] = sender_fax.fax_name
+	request["receiver_fax_name"] = receiver_fax_name
+	message.copy_properties(paper)
+	request["paper"] = message
+	requests += list(request)
+	var/msg = span_adminnotice("<b><font color=[receiver_color]>[sanitize(receiver_fax_name)] fax</font> received a message from [sanitize(sender_fax.fax_name)][ADMIN_JMP(sender_fax)]/[ADMIN_FULLMONTY(sender)]</b>")
+	to_chat(GLOB.admins, msg, confidential = TRUE)
+	for(var/client/admin in GLOB.admins)
+		if(admin.prefs.chat_toggles & CHAT_PRAYER && admin.prefs.toggles & SOUND_PRAYERS)
+			SEND_SOUND(admin, sound('sound/machines/printer.ogg'))
+
+// A special piece of paper for the administrator that will open the interface no matter what.
+/obj/item/paper/request/ui_status()
+	return UI_INTERACTIVE
```

Comment:
Is this needed? We're on GLOB.admin_state already

Reply:
Initially, I wanted to render the papers directly in the panel. But John made a complicated enough system with separation of field lists that I decided not to overcomplicate it, and just call the paper interface. Unfortunately, the papers themselves have ui_status defined to close the interface when the player isn't around. I'm not quite sure how to properly override this, so I did it this rather crude way.

Reply:
Initially, I wanted to render the papers directly in the panel. But @Timberpoes made a complicated enough system with separation of field lists that I decided not to overcomplicate it, and just call the paper interface. Unfortunately, the papers themselves have ui_status defined to close the interface when the player isn't around. I'm not quite sure how to properly override this, so I did it this rather crude way.